Dockerfile

```diff
@@ -1,22 +1,75 @@
-FROM python:3.12
+FROM ghcr.io/astral-sh/uv:python3.12-bookworm AS build-backend-stage
```

Comment:
The Dockerfile looks very professional :laughing: I wonder if it has to be so complex though.

Could we have just two stages (first would build the frontend, and the second the backend)? It seems a bit overkill to install the `apt` deps twice and use cache mounts, I don't think we need all that here. `uv` is super fast anyway.

Reply:
This first stage is there for clean runtime, First stage is also installing some dependencies that are not needed when the app is running (libsasl2-dev, libgraphviz-dev), runtime is only graphviz. 

We can remove the cache mount due to old docker build compatibility.

Reply:
Cache mount removed